packages/sites/src/get-members.ts

```diff
@@ -0,0 +1,132 @@
+ * (/community/users) endpoint. AGO will only return the subset of
+ * members which the current user has access to.
+ *
+ * @param {Array} usernames List of usernames to search for
+ * @param {object} requestOptions IHubRequestOptions
+ */
+function authenticatedGetMembers(
+  usernames: string[],
+  requestOptions: IHubRequestOptions
+): Promise<IUser[]> {
+  const urlPath = `${getPortalUrl(
+    requestOptions
+  )}/sharing/rest/community/users`;
+  const chunkSize = 100;
+  const chunkedUsernames = [];
+
+  for (let i = 0; i < usernames.length; i += chunkSize) {
+    chunkedUsernames.push(usernames.slice(i, i + chunkSize));
+  }
+
+  const chunkedOptions = chunkedUsernames.map(chunk => {
+    const q = chunk.map(username => `username:${username}`).join(" OR ");
+    return {
+      urlPath,
+      requestOptions: { params: { q, num: chunk.length }, ...requestOptions }
+    };
+  });
+
+  return batch(chunkedOptions, batchRequest).then(batchedMembers => {
+    return batchedMembers.reduce((flat: IUser[], toFlatten: IUser[]) => {
+      return flat.concat(toFlatten);
+    }, []);
+  });
+}
+
+/**
+ * fetch members individually from AGO's /community/users/{username}
+ * endpoint. This endpoint, unlike the users search endpoint which
+ * only returns the subset of members that the current user has acces
+ * to, will only limit the information returned for each member
+ * (i.e. firstname, lastname and fullname will be empty strings if
+ * an unauthenticated user tries to access a non-public profile).
+ *
+ * @param {Array} usernames List of usernames to search for
+ * @param {object} requestOptions IHubRequestOptions
+ */
+function unauthenticatedGetMembers(
+  usernames: string[],
+  requestOptions: IHubRequestOptions
+): Promise<IUser[]> {
+  return Promise.all(
+    usernames.map(username => {
+      return getUser({ username, ...requestOptions })
+        .then(response => {
+          // if the firstname, lastname, and fullname are empty strings, assume that the
+          // user is not accessible (i.e. not a public profile) and should not be returned
+          // to the unauthenticated user
+          if (response.firstName || response.lastName || response.fullName) {
+            return response;
+          }
+        })
+        .catch(e => {
+          Logger.error(
+            `Error fetching user, ${username}, from AGO user endpoint, ${e}`
+          );
+          return null;
+        });
+    })
+  ).then(members => members.filter(Boolean));
+}
+
+interface IBatchRequestOptions {
+  urlPath: "string";
+  requestOptions: IHubRequestOptions;
+}
+
+/**
+ * callback function to batch the requests to the user search endpoint
+ * if >100 usernames are supplied. This is necessary because the
+ * endpoint sets the maximum number of results to be included in the
+ * result set response to 100
+ *
+ * @param options IBatchRequestOptions
+ */
+function batchRequest(options: IBatchRequestOptions): Promise<IUser[][]> {
```

Comment:
`batchRequest` seems generic, but it's return type is specific. I'd consider either renaming to something more specific like `batchMemberRequest` and `IBatchMemberRequestOptions`, or consider making the return signature more generic. If you take the latter route, it'll require you to do some type casting where you consume this method.